ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -238,9 +242,11 @@ def getInputSpecification(cls):
         descr=r""" point/gene(s) at which crossover will occur.""")
     crossover.addSub(crossoverPoint)
     crossoverProbability = InputData.parameterInputFactory('crossoverProb', strictMode=True,
-        contentType=InputTypes.FloatType,
+        contentType=InputTypes.FloatOrStringType,
         printPriority=108,
         descr=r""" The probability governing the crossover step, i.e., the probability that if exceeded crossover will occur.""")
+    crossoverProbability.addParam("type", InputTypes.makeEnumType('crossoverProbability','crossoverProbabilityType',['static','adaptive']), True,
+                       descr="type of crossover operation to be used (e.g., static,adaptive)")
```

Comment:
Please update description for more details. For instance, give more information about what static means and what adaptive means, and state that adaptive only accepts linear (explain and give the equation) and quadratic (explain and give the equation)